bin/pt-stalk

```diff
@@ -796,84 +797,88 @@ collect() {
    local d="$1"  # directory to save results in
    local p="$2"  # prefix for each result file
 
-   local mysqld_pid=""
-   if [ ! "$OPT_MYSQL_ONLY" ]; then
-      mysqld_pid=$(_pidof mysqld | awk '{print $1; exit;}')
-   fi
+   if [ ! "$OPT_SYSTEM_ONLY" ]; then
```

Comment:
instead of `if` maybe it is good idea to create new function for system collect? and just call it in the beginning and exit if "$OPT_SYSTEM_ONLY" is set. 

And do we need new key `OPT_SYSTEM_ONLY` at all? if there is no mysqld process - just don't execute mysqld section, if OPT_MYSQL_ONLY is set don't execute the system function

Reply:
I made pt-stalk data collection more modular in commit d6e89940..623fdaec. Please review.

Regarding the option: while the main purpose for this patch is to allow pt-stalk to run on machines where MySQL is not installed, we may need to run it on machines where MySQL is installed, but not the source of the problem. For example, working with a customer who has a small MySQL server and large MongoDB or PostgreSQL that we are troubleshooting.